distribution/lib/Standard/Visualization/0.0.0-dev/src/Scatter_Plot.enso

```diff
@@ -132,10 +133,26 @@ Table.Table.axes self =
     if axes_obj.fields.size > 0 then axes_obj else Nothing
 
 ## PRIVATE
-Vector.Vector.point_data : Vector -> Object
-Vector.Vector.point_data self =
-        self.map_with_index <| i-> elem->
-            Json.from_pairs [[X.name,i],[Y.name,elem]]
+Vector.Vector.point_data : Vector -> [Int]|Nothing -> Int|Nothing-> Object
+Vector.Vector.point_data self bounds limit =
+    data = case bounds of
+        Nothing ->
+            self.map_with_index <| i-> elem->
+                Json.from_pairs [[X.name,i],[Y.name,elem]]
+        _ ->
+            minX = bounds.at 0
+            minY = bounds.at 1
+            maxX = bounds.at 2
+            maxY = bounds.at 3
+            pairs = self.map_with_index <| i-> elem->
+                if minX>i || i>maxX || minY>elem || elem>maxY then Nothing else
+                    Json.from_pairs [[X.name,i],[Y.name,elem]]
+            pairs.filter case _ of
+                Nothing -> False
+                _ -> True
+    case limit of
+        Nothing -> data
+        n -> data.take (Sample n)
```

Comment:
while I like the random sample - I think we need to include the minimum and maximum y

Reply:
I was meaning (and still think) we should include lowest and highest x value points and lowest and highest y value points within the specified bounds and then sample the remaining required number of points.

Reply:
Since [5f58a6d](https://github.com/enso-org/enso/pull/3655/commits/5f58a6d48617a7809931fb49ecbbeb4cb1373fdf) the _extreme values_ on both axis are going to be included in each sample.